File: client/src/app/_models/layoutControl.ts
export interface LayoutControl {
  id: number;
  name: string;
  description: string;
  placeholder: string | null;
  label: string;
  section: number;
  row: number;
  order: number;
  cellSource: string;
  required: boolean;
  value: string | number | Date | boolean | undefined;
}

export interface LayoutSection {
  sectionNumber: number;
  rows: LayoutRow[];
}

export interface LayoutRow {
  rowNumber: number;
  controls: LayoutControl[];
}

File: ReportJewelAPI/Entities/LayoutControl.cs
﻿using System.ComponentModel.DataAnnotations;

namespace ReportJewelAPI.Entities
{
  public class LayoutControl
  {
    public int Id { get; set; }
    public int LayoutId { get; set; }
    public Layout Layout { get; set; }
    [Required]
    public string Name { get; set; }
    public string Description { get; set; }
    public string Placeholder { get; set; }
    [Required]
    public string Label { get; set; }
    public int Section { get; set; }
    public int Row { get; set; }
    public int Order { get; set; }
    public string CellSource { get; set; }
    [Required]
    public bool Required { get; set; }
  }
}

